teku/src/test/java/tech/pegasys/teku/cli/options/ReflectionBasedBeaconRestApiOptionsTest.java

```diff
@@ -65,6 +65,66 @@ public void restApiEnabled_shouldNotRequireAValue() {
         .isEqualTo(tekuConfiguration);
   }
 
+  @Test
+  public void restApiEnabledAndPortNotSpecified_shouldProvideDefaults() {
+    TekuConfiguration tekuConfiguration = getTekuConfigurationFromArguments();
+    final BeaconRestApiConfig config = getConfig(tekuConfiguration);
+    assertThat(config.isRestApiEnabled()).isFalse();
+    assertThat(config.getRestApiPort()).isEqualTo(5051);
+    assertThat(
+            createConfigBuilder().restApi(b -> b.restApiEnabled(false).restApiPort(5051)).build())
+        .usingRecursiveComparison()
+        .isEqualTo(tekuConfiguration);
+  }
+
+  @Test
+  public void restApiEnabledNullAndPortSpecified_shouldEnable() {
+    TekuConfiguration tekuConfiguration = getTekuConfigurationFromArguments("--rest-api-port=5051");
```

Comment:
another permutation here is setting to a custom port, and not setting enabled...

When we have all the permutations so similar, it can really be worth investing in a parameteised test...